src/components/Plugin/Plugin.tsx

```diff
@@ -80,3 +90,60 @@ export function Plugin(props: PluginProps) {
     </div>
   );
 }
+
+export function PluginStore(props: PluginProps) {
+  let {
+    preIcon,
+    topAction,
+    title,
+    subtitle,
+    content,
+    variant = 'primary',
+    customClass,
+    ...rest
+  } = props;
+
+  const CLASSES: classNames = {
+    root: 'flex justify-center items-center',
+    primary: {
+      default: 'default-primary',
+      preIcon: 'bg-tertiary text-f-secondary rounded-full h-10 w-10 mb-6',
+    },
+    secondary: {
+      default: 'default-secondary',
+      preIcon: 'bg-neutral text-f-secondary rounded-full h-10 w-10 mb-6',
+    },
+  };
+
+  const VARIANT = CLASSES[variant] as classNames;
+
+  const clonedPreIcon = preIcon
+    ? cloneElement(preIcon, {
+        className: VARIANT.preIcon,
+      })
+    : null;
+
+  return (
+    <div
+      data-testid="plugin"
+      className="w-80 h-44 p-4 bg-secondary rounded-lg shadow"
+      {...rest}
+    >
+      <div className="grid grid-cols-2 h-10 mb-3">
+        <div className="flex flex-row justify-start">{clonedPreIcon}</div>
+        <div className="flex flex-row justify-end">
+          <IconContext.Provider
+            value={{ className: 'w-6 h-6 text-f-primary cursor-pointer' }}
+          >
+            {topAction}
+          </IconContext.Provider>
```

Comment:
Can you clarify for me ? is the idea to make another cloneElement  or reuse the one we have ?

Reply:
Yes it was done as expected, using the cloneElement method on the topAction, let me know if the solution satisfies you. I just had to mofiy the type from ReactNode to Jsx.Element because ReactNode was creating an arror

Reply:
> We are using `<IconContext.Provider>` to be able to set the class in the icon does not matter what the user send to us. AFAIK, we can do the same use the cloneElement.
> 
> Have a look here src/components/Input/Input.tsx lines 26-30. I guess we can reach the same...
> 
> Let me know otherwise

I see, thank you it is very clear now !